allennlp/modules/transformer/bimodal_attention.py

```diff
@@ -74,6 +111,17 @@ def _transpose_for_scores(self, x):
         x = x.view(*new_x_shape)
         return x.permute(0, 2, 1, 3)
 
+    @staticmethod
+    def _apply_mask(values: torch.FloatTensor, mask: torch.BoolTensor) -> torch.FloatTensor:
+        if len(mask.shape) == 2:
+            # We create a 4D attention mask from a 2D tensor mask.
+            # The shape is `batch_size x 1 x 1 x target_seq_len` which is broadcast
+            # to `batch_size x num_attention_heads x source_seq_len x target_seq_len`
+            mask = mask.unsqueeze(1).unsqueeze(2)
+        # `mask==1` to convert float tensors.
+        mask = (~(mask == 1)) * -10e5  # to ensure that the model also works in half-precision mode.
```

Comment:
Can you make clear in the comment which part of the code makes sure it works in half-precision?
```suggestion
        mask = (~(mask == 1)) * -10e5  # -10e5 to ensure that the model also works in half-precision mode.
```